presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -683,6 +685,36 @@ protected Scope visitQuery(Query node, Optional<Scope> scope)
                     .withParent(withScope)
                     .withRelationType(RelationId.of(node), queryBodyScope.getRelationType())
                     .build();
+
+            // create a multimap of tables in a query to columns used for each table
+            Multimap<QualifiedObjectName, String> tableColumnMap = analysis.getColumnReferenceFields().values().stream()
+                    .filter(fieldId -> {
+                        // first filter out column references that don't directly reference tables
+                        Node sourceNode = fieldId.getRelationId().getSourceNode();
+                        return sourceNode instanceof Table
+                                && analysis.getScope(sourceNode)
+                                .getRelationType()
+                                .getFieldByIndex(fieldId.getFieldIndex())
+                                .getOriginTable().isPresent();
+                    })
+                    .collect(
+                            toMultimap(
+                                    // table associated with the column
+                                    fieldId -> analysis.getScope(fieldId.getRelationId().getSourceNode())
```

Comment:
Getting scope for relation id is a temporary solution for the planner to work. In the future, this mapping will be gone, so please don't rely on it.